### DISCUSSION TITLE
Why does is_float(foo) not tell Vector that it can evaluate foo <= 0.23? (E100 issue)

### DISCUSSION THREAD

**Discussion by jerrac** [vectordotdev/vrl#752] [Q&A] [answered]
I'm experimenting with Vector and keep running into the E100 error for expressions similar to:

```
exists(.name) && .name == "load5" && exists(.gauge) && exists(.gauge.value) && is_float(.gauge.value) && .gauge.value >= 0.1
```

```
error[E100]: unhandled error
  ┌─ :1:106
  │
1 │ exists(.name) && .name == "load5" && exists(.gauge) && exists(.gauge.value) && is_float(.gauge.value) && .gauge.value >= 0.1
  │                                                                                                          ^^^^^^^^^^^^^^^^^^^
  │                                                                                                          │
  │                                                                                                          expression can result in runtime error
  │                                                                                                          handle the error case to ensure runtime success
  │
  = see documentation about error handling at https://errors.vrl.dev/#handling
  = learn more about error code 100 at https://errors.vrl.dev/100
  = see language documentation at https://vrl.dev
  = try your code in the VRL REPL, learn more at https://vrl.dev/examples

```

I understand the need to check that `.gauge.value` is actually a float and can be compared. What I don't get is why this expression doesn't work.

`exists(.gauge) && exists(.gauge.value)` makes sure it is not null, right? And `is_float(.gauge.value)` will be false if it isn't a float.

That should mean that, by the time it gets to `.gauge.value >= 0.1`, Vector should be aware that `.gauge.value` is a float.

So why is Vector throwing that E100 error?

Thanks in advance. :)

FYI, I'm testing by manually running the binary downloaded today on an Ubuntu 22.04 vagrant vm.

**Comment by jszwedko** [comment, marked as answer]
Hi @jerrac !

The issue is that the compiler isn't smart enough to realize that `.gauge.value` is a float at the point where that comparison is done. You'll need to wrap it in `float!(.gauge.value)`. https://github.com/vectordotdev/vrl/issues/91 is tracking this as an enhancement.

**Reply by jerrac** [reply]
Thanks for the answer. Hopefully that enhancement lands sometime soon. :)
